cli/compose/types/types.go

```diff
@@ -260,6 +262,15 @@ type ServiceSecretConfig struct {
 	Mode   *uint32
 }
 
+// ServiceConfigObjConfig is the config obj configuration for a service
+type ServiceConfigObjConfig struct {
```

Comment:
I'd like to experiment with extracting a common type before we merge this. I can look at it later today or tomorrow.

If it doesn't significantly remove code, I'll be ok with it like this. 

I see what you're saying about different features, but I think that's more significant for the backend. From the client, where all we're doing is reading a config and converting to other types,  I expect the code to stay quite similar.